### PR TITLE
feat: improve account syncing performance

### PR DESCRIPTION
**Pull request by mathieuartu**
## **Description**

This PR adds several performance improvements and bug fixes for [account syncing](https://www.figma.com/board/sx26FtZcQc0zs0z5hpdqBh/Account-Syncing-Solution?t=5Oy00gbHLnoPqyYP-0).

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/27330?quickstart=1)

## **Related issues**

Fixes:
- https://consensyssoftware.atlassian.net/browse/NOTIFY-1157
- https://consensyssoftware.atlassian.net/browse/NOTIFY-1158
- https://consensyssoftware.atlassian.net/browse/NOTIFY-1163

## **Manual testing steps**

1. Log in with SRP
2. Add new accounts and rename others
3. Using another browser, log in with SRP
4. Watch the magic happen!

## **Pre-merge author checklist**

- [ ] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [ ] I've completed the PR template to the best of my ability
- [ ] I’ve included tests if applicable
- [ ] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [ ] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by mathieuartu** [comment]
@metamaskbot update-policies

**Comment by mathieuartu** [comment]
@metamaskbot update-policies

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -1531,7 +1531,7 @@ export default class MetamaskController extends EventEmitter {
         },
       },
       env: {
-        isAccountSyncingEnabled: true,
+        isAccountSyncingEnabled: isManifestV3,
```
Comment (Prithpal-Sooriya) [inline app/scripts/metamask-controller.js:1534]: Yep this is accepted, we will only support syncing initailly on MV3 (much nicer UX and not UI thread blocking)

File: ui/hooks/metamask-notifications/useProfileSyncing.ts [hunk 2]
```diff
@@ -134,21 +137,32 @@ export const useAccountSyncing = () => {
...
 
   const shouldDispatchAccountSyncing = useMemo(
     () =>
-      basicFunctionality && isProfileSyncingEnabled && isUnlocked && isSignedIn,
-    [basicFunctionality, isProfileSyncingEnabled, isUnlocked, isSignedIn],
+      basicFunctionality &&
+      isProfileSyncingEnabled &&
+      isUnlocked &&
+      isSignedIn &&
+      completedOnboarding,
```
Comment (Prithpal-Sooriya) [inline ui/hooks/metamask-notifications/useProfileSyncing.ts:148]: Nice, ensure that we only run this once onboarding has completed :+1:

**Review by Prithpal-Sooriya — APPROVED** [review]

**Review by matteoscurati — APPROVED** [review]
LGTM!

**Pull request merged**